databases/week2/hyf_homework_databases_lesson2.sql

```diff
@@ -0,0 +1,257 @@
+-- Part 1: Working with tasks
+USE hyf_lesson1;
+
+insert into task (title, description, created, updated, due_date, status_id, user_id) values ('Do the homework', 'Hyf databases lesson2', NOW(),  NOW(), null, 1, 3);
+
+SELECT *
+FROM task;
+
+UPDATE task
+SET title = 'Do databases homework'
+WHERE id= 36;
+
+UPDATE task
+SET due_date = '2021-04-15 02:29:29'
+WHERE id= 36;
+
+UPDATE task
+SET status_id = 2
+WHERE id= 36;
+
+UPDATE task
+SET status_id = 3
+WHERE id= 36;
+
+DELETE
+FROM task
+WHERE id= 36;
+
+-- Part 2: School database
+CREATE DATABASE hyf_homework;
+USE hyf_homework;
+
+CREATE TABLE `class` (
+  `id` int(10) unsigned NOT NULL PRIMARY KEY AUTO_INCREMENT,
+  `name` varchar(255) NOT NULL,
+  `begins` DATETIME NOT NULL,
+  `ends` DATETIME NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE TABLE `student` (
+  `id` int(10) unsigned NOT NULL PRIMARY KEY AUTO_INCREMENT,
+  `name` varchar(255) NOT NULL,
+  `email` varchar(255) NOT NULL,
+  `phone` varchar(255) NULL,
+  `class_id` int(10) unsigned NOT NULL,
+  CONSTRAINT `fk_class` FOREIGN KEY (`class_id`) REFERENCES `class` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+CREATE INDEX idx_name
+ON student (name);
+
+ALTER TABLE class
+ADD status enum('not-started', 'ongoing', 'finished') NOT NULL;
+
+-- Part 3: More queries
+use hyf_lesson2;
+
+SELECT *
+FROM task
+JOIN user_task ON
+task.id=user_task.task_id
+JOIN user ON
+user_task.user_id=user.id
+WHERE user.email LIKE '%@spotify.com';
+
+SELECT *
+FROM task
+JOIN user_task ON
+task.id=user_task.task_id
+JOIN user ON
+user_task.user_id=user.id
+JOIN status ON
+task.status_id=status.id
+WHERE user.name = "Donald Duck" AND status.name = "Not started";
+
+SELECT *
+FROM task
+JOIN user_task ON
+task.id=user_task.task_id
+JOIN user ON
+user_task.user_id=user.id
+WHERE user.name LIKE "Maryrose%" AND month(created) = 09;
+
+SELECT monthname(created) AS month, COUNT(*) AS count
```

Comment:
This query gives me an error. Can you fix it please?

Reply:
I tried to fix, is it true now?